src/Symfony/Component/Security/Http/Logout/LogoutUrlGenerator.php

```diff
@@ -56,32 +56,24 @@ public function registerListener($key, $logoutPath, $csrfTokenId, $csrfParameter
     /**
      * Generates the absolute logout path for the firewall.
      *
-     * @param string|null $key The firewall key or null to use the current firewall key
-     *
      * @return string The logout path
      */
-    public function getLogoutPath($key = null)
+    public function getLogoutPath(?string $key = null)
     {
         return $this->generateLogoutUrl($key, UrlGeneratorInterface::ABSOLUTE_PATH);
     }
 
     /**
      * Generates the absolute logout URL for the firewall.
      *
-     * @param string|null $key The firewall key or null to use the current firewall key
-     *
      * @return string The logout URL
      */
-    public function getLogoutUrl($key = null)
+    public function getLogoutUrl(string $key = null)
```

Comment:
I would keep `?` as passing `null` explicitly is supported. We don't use `?` when passing `null` is not explicit but only a way to make an argument optional.

Reply:
IMO we should not distuish "null defaults" and "null explicit" as we can't enforce that nor does it make any difference implementation wise.